src/platform/packages/shared/kbn-alerts-ui-shared/src/common/hooks/use_get_rule_types_permissions.ts

```diff
@@ -63,19 +63,15 @@ const getFilteredIndex = ({
   return filteredIndex;
 };
 
-export const useLoadRuleTypesQuery = ({
+export const useGetRuleTypesPermissions = ({
```

Comment:
This hook was acting as the main rule types fetcher but also as privilege checker, returning computed values and not the original query. I extracted the react-query part that only fetches rule types to the new package, keeping this hook as a more privilege-checking-focused one (hence the new name). I avoided further refactors such as removing the returned rule types registry thing since there is a bit of filtering logic that might be worth studying better, and moving the hook to the new package as it's not properly a data-fetching hook (I'm wondering where these higher-level hooks should belong in our new package structure 🤔).